src/lib/string.ts

```diff
@@ -64,3 +64,39 @@ export function truncate(value: string | undefined, maxLength: number, suffix =
 
   return `${value.slice(0, maxLength)}${suffix}`;
 }
+
+/**
+ * Validation of social insurance number with checking the checksum
+ * Validation according to https://www.sozialversicherungsnummer.ch/aufbau-neu.htm
+ * @param socialInsuranceNumber The social insurance number to check
+ * Must be in one of the following formats:
+ * - "756.XXXX.XXXX.XX" with dots as seperators
+ * - "756XXXXXXXXXX" with digits only
+ * @returns The result if the social insurance number is valid or not
+ */
+export function isValidSwissSocialSecurityNumber(socialInsuranceNumber: string): boolean {
+  if (isNullOrWhitespace(socialInsuranceNumber)) {
+    return false;
+  }
+
+  const socialInsuranceNumberWithDots = new RegExp(/^756.\d{4}.\d{4}.\d{2}$/);

```

Comment:
the dot in the regex should be escaped otherwise it refers to "matches any character"

also here it might be ok to merge the 2 regex into a single one, something like this might work: 

"^756\.?\d{4}\.?\d{4}\.?\d{2}$"